File: vador/src/main/java/org/revcloud/vador/config/ValidationConfig.java
/*******************************************************************************
 * Copyright (c) 2022, salesforce.com, inc.
 * All rights reserved.
 * SPDX-License-Identifier: BSD-3-Clause
 * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
 ******************************************************************************/

/*******************************************************************************
 * Copyright (c) 2022, salesforce.com, inc.
 * All rights reserved.
 * SPDX-License-Identifier: BSD-3-Clause
 * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
 ******************************************************************************/

package org.revcloud.vador.config;

import lombok.AccessLevel;
import lombok.EqualsAndHashCode;
import lombok.Value;
import lombok.experimental.FieldDefaults;
import lombok.experimental.SuperBuilder;
import org.revcloud.vador.config.base.BaseValidationConfig;
import org.revcloud.vador.config.base.ExecutionStrategy;

@Value
@FieldDefaults(level = AccessLevel.PACKAGE)
@EqualsAndHashCode(callSuper = true)
@SuperBuilder(buildMethodName = "prepare", builderMethodName = "toValidate", toBuilder = true)
public class ValidationConfig<ValidatableT, FailureT>
    extends BaseValidationConfig<ValidatableT, FailureT> {
  ExecutionStrategy executionStrategy;
}

File: vador/src/main/kotlin/com/salesforce/vador/config/ValidationConfig.kt
/**
 * ****************************************************************************
 * Copyright (c) 2022, salesforce.com, inc. All rights reserved. SPDX-License-Identifier:
 * BSD-3-Clause For full license text, see the LICENSE file in the repo root or
 * https://opensource.org/licenses/BSD-3-Clause
 * ****************************************************************************
 */
package com.salesforce.vador.config

import com.salesforce.vador.immutables.ConfigStyle
import org.immutables.value.Value

@ConfigStyle
@Value.Immutable(copy = false)
internal abstract class AbstractValidationConfig<ValidatableT, FailureT> :
  AbstractValidationConfigSupport<ValidatableT, FailureT>() {

  abstract class Builder<ValidatableT, FailureT> :
    ConfigBuilder<ValidationConfig<ValidatableT, FailureT>>,
    ValidationBuilderDsl<ValidatableT, FailureT, ValidationConfig.Builder<ValidatableT, FailureT>>
}
